src/SqlPersistence/Saga/SqlDialect_PostgreSql.cs

```diff
@@ -3,16 +3,43 @@
 namespace NServiceBus
 {
     using System.Text;
+    using Logging;
+    using Newtonsoft.Json;
+    using Persistence.Sql;
+    using LogManager = LogManager;
 
     public partial class SqlDialect
     {
         public partial class PostgreSql
         {
+            static ILog log = LogManager.GetLogger<PostgreSql>();
+
             public override string GetSagaTableName(string tablePrefix, string tableSuffix)
             {
                 return $"\"{Schema}\".\"{tablePrefix}{tableSuffix}\"";
             }
 
+            internal override void ValidateJsonSettings(Newtonsoft.Json.JsonSerializer jsonSerializer)
+            {
+                if (!IsTypeNameHandlingCorrect(jsonSerializer))
+                {
+                    throw new SerializationException("PostgreSQL does not guarantee that properties are stored in order. As such, when using any TypeNameHandling other than 'None', then the MetadataPropertyHandling must be set to 'ReadAhead'.");
```

Comment:
it feels a bit off that we use here the SerializationException since we are not actually serializing anything. Why not just a standard exception?

Reply:
I'm not sure if that's a good argument, but it is validation closely related to serialization. May make it easier for the user to understand the source of the problem. No hard feelings around changing to generic "Exception" or "ArgumentException" maybe?

Reply:
Simon says "Also cant be a generic exception since we want a custom exception that is excluded from retries" so better leave as is IMO

Reply:
Are you saying this method is only executed during runtime and not during bootstrapping? Then it make sense